iree/turbine/kernel/wave/templates/t5_rpe_attention.py

```diff
@@ -77,22 +77,17 @@ def get_t5_rpe_attention_kernel(
     )
 
     d0, d1 = [tkw.IndexMapping.dynamic_val(i) for i in range(2)]
-    clip_upper = sympy.Piecewise(
-        (d0 - d1, d0 - d1 <= max_context_length), (max_context_length, True)
+    clip = sympy.Piecewise(
+        (d0 - d1, (d0 - d1 <= max_context_length) & (d0 - d1 > 0)), (0, True)
     )
-    clip_lower_and_upper = sympy.Piecewise((clip_upper, d0 - d1 >= 0), (0, True))
     offset_mapping = tkw.IndexMapping(
         num_iterators=2,
-        inputs={M: i, K2: clip_lower_and_upper},
+        inputs={M: i, K2: clip},
         outputs={M: i, K2: j},
-        dynamic_val_mappings=({M: i, K2: j}, {M: i, K2: j}),
+        dynamic_val_mappings=({M: i}, {K2: j}),
```

Comment:
they should both be of shape K2 and iterator j

Reply:
I think this was resolved from offline discussion, lmk if we want to fix